lib/cubits/buy_tickets_page/buy_tickets_cubit.dart

```diff
@@ -1,34 +1,26 @@
 import 'package:bloc/bloc.dart';
-import 'package:coffeecard/errors/network_exception.dart';
-import 'package:coffeecard/generated/api/coffeecard_api.swagger.swagger.dart';
+import 'package:coffeecard/data/repositories/v1/product_repository.dart';
+import 'package:coffeecard/models/ticket/product.dart';
 import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 part 'buy_tickets_state.dart';
 
+enum FilterCategory { clipCards, singleTickets }
+
 class BuyTicketsCubit extends Cubit<BuyTicketsState> {
-  BuyTicketsCubit() : super(const BuyTicketsLoading());
+  final ProductRepository _repository;
 
-  Future<void> getTickets() async {
-    try {
-      emit(const BuyTicketsLoading());
-      await Future.delayed(const Duration(seconds: 1));
+  BuyTicketsCubit(this._repository) : super(const BuyTicketsLoading());
 
-      //FIXME: uncomment when data is available
-      //final List<ProductDto> tickets = await _productRepository.getProducts();
-      final tickets = [
-        ProductDto(
-          id: 1,
-          price: 10,
-          numberOfTickets: 10,
-          name: 'Test Coffee',
-          description: 'This is a test',
-        )
-      ];
+  Future<void> getTickets() async {
```

Comment:
By the naming of this cubit, I thought this cubit was only meant to be used by the "Buy Tickets" flow.
I'm in favour of making this work for both flows, but the implementation of the single drink page has been postponed, so does it make sense to wait with that and focus on how it's working for the "Buy tickets" flow only?

Reply:
I agree that the naming of the cubit is weird. A better name would likely be product_cubit, since that is what it is dealing with. It does not handle the purchasing of items after all. The method name should also be renamed, since it is not getting tickets, but products. 

As for postponing the retrieval of single products, well, I suppose we can. We haven't really discussed how we want to handle the buy one drink flow recently, either. We could do it with our current endpoints, it would just take an extra request.